### PR TITLE
🧭 Discover mixpanel analytics

### PR DESCRIPTION
**Pull request by piotrkonowrocki**
# [🧭 Discover mixpanel analytics](https://app.shortcut.com/oazo-apps/story/6185/discover-oasis-analytics)

All events are based on this sheet https://docs.google.com/spreadsheets/d/1Jie1SX_5i4-jDZLMKnCUkq4lhchkPKonIOVPNNJtsA8/edit#gid=308180383 with an exception of `ScrollTable` event which will be implemented differently, as a separated story.
  
## Changes 👷‍♀️

- Added dedicated events for tracking user behavior on Discover Oasis pages.
  
## How to test 🧪

- Try to trigger all trackable actions on Discover page UI, then look for request to `/api/t` and compare sent parameters with those required by Google sheet,
- please note that all parameters in `eventBody` were intentioanlly changed to camelCase and all event names to kebabCase.

### CODE REVIEW

File: analytics/analytics.ts [hunk 1]
```diff
@@ -866,6 +882,48 @@ export const trackingEvents = {
       mixpanelInternalAPI(EventTypes.ButtonClick, eventBody)
     },
   },
+
+  discover: {
+    selectedInNavigation: (userContext: MixpanelUserContext) => {
+      mixpanelInternalAPI('btn-click', {
```
Comment (marcinciarka) [inline analytics/analytics.ts:888]: Use `EventTypes` enum from :70

**Review by marcinciarka — CHANGES_REQUESTED** [review]

File: features/discover/views/DiscoverView.tsx [hunk 2]
```diff
@@ -1,14 +1,24 @@
...
+  const { context$ } = useAppContext()
+  const [context] = useObservable(context$)
+
+  const userContext = getMixpanelUserContext(i18n.language, context)
+
   return (
     <DiscoverWrapperWithIntro key={kind}>
-      <DiscoverNavigation kind={kind} />
-      <DiscoverControl kind={kind} />
+      <DiscoverNavigation kind={kind} userContext={userContext} />
```
Comment (johnnieskywalker) [inline features/discover/views/DiscoverView.tsx:20]: image
image
image

image

image

Seems all params are ok but Page is missing

**Review by johnnieskywalker — CHANGES_REQUESTED** [review]

File: analytics/analytics.ts [hunk 3]
```diff
@@ -866,6 +882,48 @@ export const trackingEvents = {
...
+        id: 'Discover',
```
Comment (marcinciarka) [inline analytics/analytics.ts:889]: Theres `Page` name missing in all of these events (column B in the mixpanel event sheets).

**Review by marcinciarka — CHANGES_REQUESTED** [review]

File: analytics/analytics.ts (hunk 1 above)
Comment (piotrkonowrocki) [reply analytics/analytics.ts]: Fixed.

File: features/discover/views/DiscoverView.tsx (hunk 2 above)
Comment (piotrkonowrocki) [reply features/discover/views/DiscoverView.tsx:20]: For whatever wrong reason I somehow skipped first three params, `product`, `page` and `section`. All done now.

File: analytics/analytics.ts (hunk 3 above)
Comment (piotrkonowrocki) [reply analytics/analytics.ts:898]: See: https://github.com/OasisDEX/oasis-borrow/pull/1592#discussion_r1001545892
Same comment applies.

File: analytics/analytics.ts (hunk 1 above)
Comment (marcinciarka) [reply analytics/analytics.ts]: 👌

**Review by marcinciarka — APPROVED** [review]

**Review by johnnieskywalker — APPROVED** [review]

**Pull request merged**